### DISCUSSION TITLE
Errors when running pre-commit

### DISCUSSION THREAD

**Discussion by tm-jc-nacpil** [thinkingmachines/geowrangler#93] [Q&A] [answered]
Hi! I encountered the following error when running `poetry run pre-commit run -a` for the first time. 

![Uploading image.png…]()

```
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('go', 'install', './...')
return code: 1
expected return code: 0
stdout:
    Executable `go` not found
stderr: (none)
```

**Comment by butchtm** [comment, marked as answer]
@jtmiclat 
Installing go is a requirement due to the [gitleaks plugin](https://github.com/zricethezav/gitleaks) in pre-commit. 

Should we add this in the setup instructions or do we disable the gitleaks plugin in the pre-commit?

**Reply by jtmiclat** [reply]
Gitleaks has a docker alternative pre-commit  https://github.com/zricethezav/gitleaks/blob/master/.pre-commit-hooks.yaml see `gitleaks-docker` which might be more user-friendly than to install golang.

Should we require/suggest users install docker?

We need ruby already and docker seems to be easier to install than ruby for the Jekyll docs.  Jekyll requires a certain version of ruby and is a bit unclear how to pin this Jekyll version. So rather than installing ruby + go, we can have them install docker instead. 


If we dont require users to install docker, I'm alright removing it from pre-commit but adding to a linting gh action.

**Reply by butchtm** [reply]
My suggestion is we try removing it from pre-commit for now (Ok with adding a linting gh action). 

Installing jekyll is optional (viewing the doc site is optional since the PR triggers a preview site) but the pre-commit is required, so if we want to implement pre-commit then its much simpler to remove the go requirement.

**Reply by butchtm** [reply]
Addressed the gitleaks requirement (removed it from pre-commit) and updated the README  dev setup instructions in [this PR](https://github.com/thinkingmachines/geowrangler/pull/104).

**Comment by joshuacortez** [comment]
Closing this since we're not using pre-commits in the current version
